apm-sniffer/optional-plugins/trace-ignore-plugin/src/main/java/org/apache/skywalking/apm/plugin/trace/ignore/matcher/FastPathMatcher.java

```diff
@@ -122,9 +138,11 @@ private boolean wildcardMatch(String pat, int p, String str, int s) {
     }
 
     private boolean multiWildcardMatch(String pat, int p, String str, int s) {
-        // End of pattern, just check the end of string is '/' quickly.
-        if (p >= pat.length() && s < str.length()) {
-            return str.charAt(str.length() - 1) != '/';
+        switch (safeCharAt(pat, p)) {
+            // End of pattern, just return true quickly.
+            case 0: return true;
+            // Skip next '/' for pattern to match zero path part.
+            case '/': p++;
```

Comment:
Add this branch to support `eureka/**/test` matches `eureka/test` (** means zero path part)

Reply:
> OK, so, this becomes a hybrid PR. @DominikHubacek Please recheck this PR, and if you don't feel comfortable about this PR will(if merged) be on your name, we could consider other options.

its ok, I checked if my changes still work as intended and fixed 1 test based on devkanro comment